### ISSUE TITLE
Port Resolving Support in Spring-Data-Redis for Jedis

### ISSUE THREAD

**Issue by anish8129** [spring-projects/spring-data-redis#2781]
I am trying to Test Redis Cluster Using TestContaienrs.
I have used `lettuce` to test but that which actually required Manual NAT port mapping, because of implementation details of the Redis cluster node discover algorithm. Redis Cluster uses gossip in order to auto-discover nodes. Each cluster node announces its IP and port, and the application uses these IPs and PORTs for updating cluster topology(io.lettuce.core.cluster.topology.DefaultClusterTopologyRefresh). So, the application has to be able to connect to these addresses.

Here is an example: [RedisConfig.java](https://github.com/Hixon10/spring-redis-cluster-testcontainers/blob/master/src/test/java/ru/hixon/springredisclustertestcontainers/RedisConfig.java)

I want to have similar support for jedis as well in Spring-Data-Redis.
After going through the code I noticed this `redis.clients.jedis.HostAndPortMapper` class which we could use in the `org.springframework.data.redis.connection.jedis.JedisClientConfiguration`.

We need to add this new contract in the `JedisClientConfiguration`
```
	/**
	 * @return the optional {@link HostAndPortMapper}.
	 */
	Optional<HostAndPortMapper> getHostAndPortMapper();
```
This will be implemented by the default `DefaultJedisClientConfigurationBuilder`

I have tried this change on my local and tested as well.
Here is the sample PR which is in my forked branch
https://github.com/spring-projects/spring-data-redis/pull/2780

Please suggest a better way to handle this in jedis if it already exist.

**Comment by junghoon-vans** [comment]
If connecting to a Redis cluster, you have to set up `RedisClusterConfiguration`. Have you tried that method?

**Comment by junghoon-vans** [comment]
When connecting to a Redis cluster, you can use `RedisClusterConfiguration`. 
Have you tried the following methods?

**Comment by anish8129** [comment]
> When connecting to a Redis cluster, you can use `RedisClusterConfiguration`. Have you tried the following methods?

Yes I have used `RedisClusterConfiguation` when I am connecting to redis cluster.
here is snippet link from the example project whhich i have mentioned in the description.
https://github.com/anish8129/prom-service/blob/4caefa13d23728d80058a34b0b36aa5d8a12fe00/src/main/java/com/redis/util/RedisBeanUtils.java#L60-L67
